synapse/handlers/room_member.py

```diff
@@ -515,10 +514,17 @@ async def update_membership_locked(
         elif effective_membership_state == Membership.LEAVE:
             if not is_host_in_room:
                 # perhaps we've been invited
-                invite = await self.store.get_invite_for_local_user_in_room(
-                    user_id=target.to_string(), room_id=room_id
-                )  # type: Optional[RoomsForUser]
-                if not invite:
+                (
+                    current_membership_type,
+                    current_membership_event_id,
+                ) = await self.store.get_local_current_membership_for_user_in_room(
+                    target.to_string(), room_id
+                )
+                if (
+                    not current_membership_type
+                    or current_membership_type != Membership.INVITE
```

Comment:
I don't think we need to check if it is None first since `None != Membership.INVITE` anyway.